File: client/cs/src/Proton.Drive.Sdk/Nodes/PhotoTagsUpdate.cs
namespace Proton.Drive.Sdk.Nodes;

public sealed record PhotoTagsUpdate
{
    public required NodeUid NodeUid { get; init; }

    public required IReadOnlyList<PhotoTag> TagsToAdd { get; init; }

    public required IReadOnlyList<PhotoTag> TagsToRemove { get; init; }
}

File: incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/extension/PhotoTagsUpdate.kt
package me.proton.drive.sdk.extension

import me.proton.drive.sdk.entity.PhotoTagsUpdate
import proton.drive.sdk.ProtonDriveSdk
import proton.drive.sdk.photoTagsUpdate

fun PhotoTagsUpdate.toProto(): ProtonDriveSdk.PhotoTagsUpdate = photoTagsUpdate {
    nodeUid = this@toProto.nodeUid.value
    tagsToAdd += this@toProto.tagsToAdd.map { it.toSdkPhotoTag() }
    tagsToRemove += this@toProto.tagsToRemove.map { it.toSdkPhotoTag() }
}
